src/emulator/dataconnectEmulator.ts

```diff
@@ -160,20 +176,38 @@ export class DataConnectEmulator implements EmulatorInstance {
   }
 
   private async discoverRunningInstance(): Promise<boolean> {
-     const emuInfo = await this.emulatorClient.getInfo();
-     if (!emuInfo) {
+    const emuInfo = await this.emulatorClient.getInfo();
+    if (!emuInfo) {
       return false;
-     }
-     const serviceInfo = await load(this.args.projectId, this.args.locationId,this.args.configDir);
-     const sameService =  emuInfo.services.some(s => serviceInfo.dataConnectYaml.serviceId === s.serviceId);
-     if (!sameService) {
-      throw new FirebaseError(`There is a Data Connect emulator already running on ${
-        this.args.listen[0].address
-      }:${
-        this.args.listen[0].port
-      }, but it is emulating a different service. Please stop that instance of the Data Connect emulator, or specify a different port in 'firebase.json'`);
-     }
-     return true;
+    }
+    const serviceInfo = await load(this.args.projectId, this.args.locationId, this.args.configDir);
+    const sameService = emuInfo.services.some(
+      (s) => serviceInfo.dataConnectYaml.serviceId === s.serviceId,
+    );
+    if (!sameService) {
+      throw new FirebaseError(
+        `There is a Data Connect emulator already running on ${this.args.listen[0].address}:${this.args.listen[0].port}, but it is emulating a different service. Please stop that instance of the Data Connect emulator, or specify a different port in 'firebase.json'`,
+      );
+    }
+    return true;
+  }
+
+  private watchUnmanagedInstance() {
+    return setInterval(async () => {
+      if (!this.usingExistingEmulator) {
+        return;
+      }
+      const emuInfo = await this.emulatorClient.getInfo();
+      if (!emuInfo) {
+        this.logger.logLabeled(
+          "INFO",
+          "dataconnect",
+          "The already running emulator seems to have shut down. Starting a new instance of the Data Connect emulator...",
+        );
+        // If the other emulator was shut down, we spin our own copy up
+        await this.start();
```

Comment:
Going to leave this for now - with a 5s interval, there is very little concern about multiple calls from the same process overlapping.

Reply:
5s is not guaranteed. If a previous timeout callback is delayed, it may queue up with the next call and two callbacks can be dequeued from the event queue right next to each other.

Normally, I wouldn't even pick on stuff like this, but the CLI has a lot of readFileSync etc that blocks the main thread, making it quite likely to happen. I'd push for at least a TODO comment.